exonum-java-binding/core/rust/integration_tests/tests/service_runtime.rs

```diff
@@ -19,21 +19,51 @@ extern crate integration_tests;
 extern crate java_bindings;
 #[macro_use]
 extern crate lazy_static;
-extern crate parking_lot;
 
 use exonum_testkit::TestKitBuilder;
 use integration_tests::{
     fake_service::*,
     vm::{get_fake_service_artifact_path, get_fakes_classpath},
 };
-use java_bindings::{Config, InternalConfig, JavaServiceRuntime, JvmConfig, RuntimeConfig};
-use parking_lot::{Mutex, MutexGuard};
+use java_bindings::{
+    jni::JavaVM, Config, InternalConfig, JavaServiceRuntime, JvmConfig, RuntimeConfig,
+};
+use std::sync::{
+    atomic::{AtomicUsize, Ordering, ATOMIC_USIZE_INIT},
+    Arc,
+};
 
 lazy_static! {
-    // Reference to the JavaServiceRuntime is wrapped with Mutex in order to execute tests (more
-    // precise - Java code for artifacts loading and services creation) sequentially.
-    // We use Mutex from the `parking_lot` crate because it doesn't get poisoned in case of panic.
-    static ref SERVICE_RUNTIME: Mutex<JavaServiceRuntime> = Mutex::new(create_runtime());
+    static ref VM: Arc<JavaVM> = {
```

Comment:
Why are all these empty parameters needed? Is there _anything_ happening except passing the classpath?

Reply:
Creation of JVM with configs is moved to a separate test in a separate file.